gitlab/gitlab_exporter.py

```diff
@@ -118,15 +127,52 @@ def request_entities_from_gitlab_using_pagination(api_url: str, headers: dict, p
             else:
                 request_more_entities = False
         else:
-            print(f"Failed to retrieve projects, Page Number: {current_page}, Status code: {response.status_code}")
+            print(f"Failed to retrieve entities, Page Number: {current_page}, Status code: {response.status_code}")
     
     return entities_from_gitlab
 
+def get_all_projects_from_gitlab():
+    created_projects_in_port = 0
+
+    # By default, in gitlab projects a request returns 20 results at a time because the API results are paginated.
+    # Archived projects are included by default so we need to exclude them.
+    projects_from_gitlab = request_entities_from_gitlab_using_pagination(
+        f"{GITLAB_GROUP_URL}/projects",
+        {},
+        {'include_subgroups': True, 'archived': False}
+    )
+
+    print(f"Found {len(projects_from_gitlab)} projects in GitLab")
+
+    for project in projects_from_gitlab:
+        projects_ids_to_names[project['id']] = project['path_with_namespace'].replace('/', '-')
+
+    # Creates microservices in Port
+    if len(projects_from_gitlab) > 0:
```

Comment:
Is it possible that a project id will have a character we dont support in our identifier?